cdci_data_analysis/analysis/instrument.py

```diff
@@ -716,6 +731,32 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
 
         return has_prods
 
+    def upload_files_request(self, par_dic, request, upload_dir, products_url, token=None):
+        list_uploaded_files = []
+        if request.method == 'POST':
+            if validators.url(products_url):
+                basepath = os.path.join(products_url, 'dispatch-data/download_file')
+            else:
+                basepath = os.path.join(products_url, 'download_file')
+            for f in request.files:
+                # TODO needed since those two files are extracted in a previous step
+                if f != 'user_scw_list_file' and f != 'user_catalog_file':
+                    f_path = upload_file(f, upload_dir)
+                    if f_path is not None:
+                        file_hash = make_hash_file(f_path)
+                        f_name, f_ext = os.path.splitext(os.path.basename(f_path))
+                        new_file_name = f_name.split('.')[0] + '_' + file_hash + f_ext
+                        new_file_path = os.path.join(upload_dir, new_file_name)
+                        list_uploaded_files.append(new_file_name)
+                        os.rename(f_path, new_file_path)
+                        dpars = urlencode(dict(file_list=new_file_name,
+                                               return_archive=False,
+                                               token="INSERT_YOUR_TOKEN_HERE"))
+                        download_file_url = f"{basepath}?{dpars}"
+                        par_dic[f] = download_file_url
+        return list_uploaded_files
+
```

Comment:
so this stores hash, not url, right? I think we discussed URL? So that it's universally valid.

Reply:
the `list_uploaded_files` list stores the filenames, to be then used to update the ownership json file, inside the directory maintaining the uploaded files

the urls are then part of the `par_dic` dict